### ISSUE TITLE
D2 ocean fails to build if ldmd2 is used as the compiler

### ISSUE THREAD

**Issue by joseph-wakeling-sociomantic** [sociomantic-tsunami/ocean#496]
Current D2 ocean builds with recent dmd releases (e.g. 2.078.x, 2.079.x).  However, if `ldmd2` (the DMD-alike LDC frontend) is used, then unittests fail with the following message:

```
./src/ocean/math/IEEE.d(293): Error: unimplemented return type IeeeFlags for implicit abi return
```

The line in question is an `asm` block, which suggests that ocean's `asm` sections will need adaptation or re-implementation if they are to be supported by LDC.

This is a real shame, as we have some use-cases where LDC's much better/faster number-crunching would be a real benefit.

**Comment by mihails-strasuns-sociomantic** [comment]
Ocean has never supported neither LDC nor GDC. It is only planned for somewhat long-distance future.

**Comment by david-eckardt-frequenz** [comment]
> The line in question is an asm block, which suggests that ocean's asm sections will need adaptation or re-implementation if they are to be supported by LDC.

Actually no. The line in question affects the D function containing the asm section. The function body doesn’t have a `return` statement so the compiler needs to know how to return the result implicitly. DMD seems to know what to do to implicitly return a struct with one `int` member while LDC doesn’t. LDC can be helped by returning `int` instead; fortunately, this is a `private` function so the public interface won’t change.

Fix submitted: #714

**Comment by joseph-wakeling-frequenz** [comment]
Someone remove the `res-wontfix` tag, the fix is now merged ;-)
